### PR TITLE
Add state transition logic for fault state

### PR DESCRIPTION
**Pull request by pasitlaothamatas**
### Summary

- add state transition from fault state->init state
- add error table to the world detect air shdn error in the fault state

### Testing Done

- unit tests for BMS-29 and BMS-30

### Checklist
*Please change `[ ]` to `[x]` when you are ready.*
- [ ] I have read and followed the code conventions detailed in [README.md](../README.md) (*This will save time for both you and the reviewer!*).
- [ ] If this pull request is longer then **500** lines, I have provided *explicit* justification in the summary above explaining why I *cannot* break this up into multiple pull requests (*Small PR's are faster and less painful for everyone involved!*).

### CODE REVIEW

File: boards/BMS/Src/App/states/App_PreChargeState.c [hunk 1]
```diff
@@ -2,6 +2,7 @@
 
 #include "App_SetPeriodicCanSignals.h"
 #include "App_SharedMacros.h"
+#include "App_SharedErrorTable.h"
```
Comment (alxshev) [inline boards/BMS/Src/App/states/App_PreChargeState.c:5]: why the include?

File: boards/BMS/Test/Src/Test_StateMachine.cpp [hunk 2]
```diff
@@ -849,4 +853,64 @@ TEST_F(
         App_CanTx_GetPeriodicSignal_STATE(can_tx_interface));
 }
```
Comment (alxshev) [inline boards/BMS/Test/Src/Test_StateMachine.cpp:855]:
```suggestion

// BMS-29
```

File: boards/BMS/Test/Src/Test_StateMachine.cpp [hunk 3]
```diff
@@ -849,4 +853,64 @@ TEST_F(
...
+    // Close AIR+ to avoid false positives
+    App_CanTx_SetPeriodicSignal_AIR_POSITIVE(
+        can_tx_interface, CANMSGS_BMS_AIR_STATES_AIR_POSITIVE_CLOSED_CHOICE);
+
+    SetInitialState(App_GetFaultState());
+    ASSERT_EQ(
+        CANMSGS_BMS_AIR_STATES_AIR_POSITIVE_OPEN_CHOICE,
+        App_CanTx_GetPeriodicSignal_AIR_POSITIVE(can_tx_interface));
+}
+
```
Comment (alxshev) [inline boards/BMS/Test/Src/Test_StateMachine.cpp:867]:
```suggestion

// BMS-30
```

File: boards/BMS/Test/Src/Test_StateMachine.cpp [hunk 4]
```diff
@@ -849,4 +853,64 @@ TEST_F(
...
+        CANMSGS_BMS_STATE_MACHINE_STATE_FAULT_CHOICE,
+        App_CanTx_GetPeriodicSignal_STATE(can_tx_interface));
+
+    is_air_negative_closed_fake.return_val = false;
+    LetTimePass(state_machine, 1000);
+    ASSERT_EQ(
+        CANMSGS_BMS_STATE_MACHINE_STATE_INIT_CHOICE,
+        App_CanTx_GetPeriodicSignal_STATE(can_tx_interface));
+}
+
```
Comment (alxshev) [inline boards/BMS/Test/Src/Test_StateMachine.cpp:887]:
```suggestion

// BMS-30
```

**Review by alxshev — APPROVED** [review]

File: boards/BMS/Src/App/states/App_PreChargeState.c (hunk 1 above)
Comment (pasitlaothamatas) [reply boards/BMS/Src/App/states/App_PreChargeState.c:5]:
```suggestion
```

Comment (pasitlaothamatas) [reply boards/BMS/Src/App/states/App_PreChargeState.c]: removed

**Pull request merged**